crates/kubelet/src/provider.rs

```diff
@@ -50,7 +50,7 @@ pub trait Provider {
     const ARCH: &'static str;
 
     /// Allows provider to populate node information.
-    fn node(&self, _builder: &mut NodeBuilder) {}
+    async fn node(&self, _builder: &mut NodeBuilder) {}
```

Comment:
By providing a default  implementation we're saying that usually people won't need to override this yet we currently override it in both providers. Are we sure we want to default to this implementation?

Reply:
I actually think it is a good default for making this more generic for others. I think that other providers shouldn't necessarily need extra configuration. We can always remove it down the road